GameOfRunes/GameOfRunes/GameEngine/Components/HealthComponent.swift

```diff
@@ -10,10 +10,28 @@ import SpriteKit
 import GameplayKit
 
 class HealthComponent: GKComponent {
-    var healthPoints: Int
+    private weak var healthBarNode: HealthBarNode?
```

Comment:
Isnt it weak?

Reply:
hmmmm would there be a case where the node is removed from the parent but the entity is still in the game engine? Because if there is such a case it would cause bugs since the node would be deallocated

Reply:
Hmm unlikely, because healthBar belongs to the player and it will always exist during gameplay.